fn/set.go

```diff
@@ -31,6 +31,16 @@ func (s Set[T]) Contains(e T) bool {
 	return ok
 }
 
+// IsEmpty returns true if the set is empty.
+func (s Set[T]) IsEmpty() bool {
+	return len(s) == 0
+}
+
+// Size returns the number of elements in the set.
+func (s Set[T]) Size() int {
+	return len(s)
+}
```

Comment:
Cast this to a uint, there is no reason for negative sizes. I have no idea why golang made the decision to use ints here but that decision was wrong.